### PR TITLE
Gutenboarding: Always enroll in FSE when creating a site

### LINKED ISSUE #60253: FSE Rollout: remove opt-in screen from the /new flow
### Details

The goal of this task is to default all English-speaking existing customers to FSE. It's currently already at 100% but we are still showing an opt-in screen that needs to be removed. While doing this we need to be mindful of preserving the existing stats tracking.

### Checklist

- [x] Remove the opt-in screen, everyone gets FSE beta by default
- [x] Ensure that the FSE beta tracks event is preserved for everyone that enters the flow

### Related

_No response_

### PR DESCRIPTION
**Pull request by jeyip**
The original PR had to be reverted because of failing canary tests. This is a copy. https://github.com/Automattic/wp-calypso/pull/60271

#### Changes proposed in this Pull Request

General clean-up of Gutenboarding, removing everything related to FSE opt-in and forcing `enable_fse` in the new site request `options` object.

It also lets the user pick a classic AnchorFM theme if necessary, following the discussion in p1642621207111500-slack-CHN6J22MP.

Closes #60253.

#### Testing instructions

Create a site through `/new`. You'll notice only Universal Themes show up in the design picker and that you're FSE eligible as soon as the site is created.

Creating a site by having [AnchorFM as a referer](pdgAPm-1G-p2) displays only Anchor classic themes in the design picker, even though the site will get enrolled in FSE.

#### Next steps

Merge [this commit](https://github.com/Automattic/wp-calypso/commit/28c45a937dd52f6f72ffb8127bd58e9375724834) putting Anchor themes at the top while displaying Universal Themes below them. Reference:  p1642633206135400/1642621207.111500-slack-CHN6J22MP

### CODE REVIEW

**Comment by worldomonation** [comment]
I'm going to trigger the Pre-Release tests manually in TeamCity ([you can see it here](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Pre_Release?branch=remove%2Ffse-beta-opt-in-new-second-attempt&buildTypeTab=overview&mode=builds)) to verify the change to Quadrat worked.

**Comment by jeyip** [comment]
> I'm going to trigger the Pre-Release tests manually in TeamCity (you can see it here) to verify the change to Quadrat worked.

Thanks again for chiming in @worldomonation. I'm seeing more failures, but this is helpful! Do you know if there's documentation about how to run the pre-release tests manually to debug these issues?

**Comment by worldomonation** [comment]
Ah, the failure is probably because the font pairing screen has been removed with FSE beta?

As for instructions on how to run Pre-Release tests (or any of the CI tasks, really) manually is something I need to write up in the Field Guide. I place higher priority on this task.

For now, what you can do is:

1. Click on 'Details' for one of the tasks, eg. `E2E Tests (desktop ) (Web app)`.
2. When TeamCity loads, locate the menu at the top 
image
3. Click on the `Web app` cookie crumb.
4. You should see a list of build configurations like this:
image
5. Click on 'Run'

**Comment by worldomonation** [comment]
@jeyip Yeah, I checked the `live` site and looks like the Font Pairing screen is also gone. In this case, it would suffice to simply remove the step entirely.

Manually stepping through the revised Gutenboarding flow, it looks like everything else should work.

Edit: Looks like the step `Lands in Home Dashboard` has also changed.

File: test/e2e/specs/specs-playwright/wp-gutenboarding__happy-path.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ describe( DataHelper.createSuiteTitle( 'Gutenboarding: Create' ), function () {
 		} );
 
 		it( 'Select Vesta as the site design', async function () {
```
Comment (worldomonation) [inline test/e2e/specs/specs-playwright/wp-gutenboarding__happy-path.ts]: It still references Vesta here.

**Review by worldomonation — APPROVED** [review]

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx [hunk 2]
```diff
@@ -156,35 +89,27 @@ const Designs: React.FunctionComponent = () => {
...
 		// any more and `getRandomizedDesigns` is auto-populated in a state-agnostic way.
 		const availableDesigns = getAvailableDesigns( {
-			useFseDesigns: isEnrollingInFseBeta,
+			featuredDesignsFilter: ( design ) => {
+				if ( isAnchorFmSignup ) {
+					return design.features.includes( 'anchorfm' );
+				}
+
+				return !! design.is_fse;
+			},
```
Comment (stephanethomas) [inline client/landing/gutenboarding/onboarding-block/designs/index.tsx:98]: This change triggers the following TypeScript errors:

```javascript
yarn tsc -p client/tsconfig.json --noEmit

client/landing/gutenboarding/onboarding-block/designs/index.tsx:92:4 - error TS2345: Argument of type '{ featuredDesignsFilter: (design: any) => any; randomize: true; }' is not assignable to parameter of type 'AvailableDesignsOptions'.
  Object literal may only specify known properties, and 'featuredDesignsFilter' does not exist in type 'AvailableDesignsOptions'.

 92    featuredDesignsFilter: ( design ) => {
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 93     if ( isAnchorFmSignup ) {
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
 97     return !! design.is_fse;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 98    },
    ~~~~

client/landing/gutenboarding/onboarding-block/designs/index.tsx:92:29 - error TS7006: Parameter 'design' implicitly has an 'any' type.

92    featuredDesignsFilter: ( design ) => {
                               ~~~~~~
```

Could you fix them when you have a few @jeyip? Thanks!

Comment (jeyip) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx:98]: Of course! As a side note, I didn't draft the [original PR](https://github.com/Automattic/wp-calypso/pull/60271). I was shepherding someone else's work into production. I'm more than happy to take a look though.

cc @zaguiini

Comment (stephanethomas) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx:98]: Well I've got a good news, apparently you won't have to do anything 🙂. This is an issue with `tsc`, I just needed to rebuild packages and those errors went away (more information [here](https://github.com/Automattic/wp-calypso/pull/60274#discussion_r790978873)).

Comment (jeyip) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx:98]: Thanks for the update 🙂

**Pull request merged**